src/DotNetWorker.Core/FunctionMetadata/DefaultMetadataProviderAttribute.cs

```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.Functions.Worker.Core.FunctionMetadata;
+
+[AttributeUsage(AttributeTargets.Class)]
+public sealed class DefaultMetadataProviderAttribute : Attribute
```

Comment:
Currently using an attribute to mark the default internal one. I went back and forth on this one but at the end I ended up adding this attribute to the Core project. Maybe someone has a better idea?